src/knot_protocol.c

```diff
@@ -98,42 +98,29 @@
 
 #define TYPE_ID_MASK(type)	(type & 0x000F)
 
-static const struct schema {
-	uint8_t		value_type;
-	uint32_t	unit_mask;
-} basic_types[]  = {
-	{KNOT_VALUE_TYPE_RAW, 0xFFFF			},
-	{KNOT_VALUE_TYPE_INT, UNIT_VOLTAGE_MASK		},
-	{KNOT_VALUE_TYPE_INT, UNIT_CURRENT_MASK		},
-	{KNOT_VALUE_TYPE_INT, UNIT_RESISTANCE_MASK	},
-	{KNOT_VALUE_TYPE_INT, UNIT_POWER_MASK		},
-	{KNOT_VALUE_TYPE_INT, UNIT_TEMP_MASK		},
-	{KNOT_VALUE_TYPE_INT, UNIT_HUM_MASK		},
-	{KNOT_VALUE_TYPE_INT, UNIT_LUM_MASK		},
-	{KNOT_VALUE_TYPE_INT, UNIT_TIME_MASK		},
-	{KNOT_VALUE_TYPE_INT, UNIT_MASS_MASK		},
-	{KNOT_VALUE_TYPE_INT, UNIT_PRESSURE_MASK	},
-	{KNOT_VALUE_TYPE_INT, UNIT_DISTANCE_MASK	},
-	{KNOT_VALUE_TYPE_FLOAT, UNIT_ANGLE_MASK		},
-	{KNOT_VALUE_TYPE_FLOAT, UNIT_VOL_MASK		},
-	{KNOT_VALUE_TYPE_FLOAT, UNIT_AREA_MASK		},
-	{KNOT_VALUE_TYPE_FLOAT, UNIT_RAIN_MASK		},
-	{KNOT_VALUE_TYPE_FLOAT, UNIT_DENSITY_MASK	},
-	{KNOT_VALUE_TYPE_FLOAT, UNIT_LAT_MASK		},
-	{KNOT_VALUE_TYPE_FLOAT, UNIT_LONG_MASK		},
-	{KNOT_VALUE_TYPE_INT, UNIT_SPEED_MASK		},
-	{KNOT_VALUE_TYPE_FLOAT, UNIT_VOLFLOW_MASK	},
-	{KNOT_VALUE_TYPE_INT, UNIT_ENERGY_MASK		}
-};
-
-static const uint8_t logic_types[]  = {
-	KNOT_VALUE_TYPE_BOOL,			// KNOT_TYPE_ID_PRESENCE
-	KNOT_VALUE_TYPE_BOOL,			// KNOT_TYPE_ID_SWITCH
-	KNOT_VALUE_TYPE_RAW			// KNOT_TYPE_ID_COMMAND
-};
```

Comment:
I think we can't tie the value type with the semantic, because there are several use cases that we were unable to map. So, I believe that we can leave the responsibility to the user.

Reply:
I agree with you @vitbaq, when talking about the responsibility. Also, the discussion could be summarized in this question: how strict will be the protocol defined by us?
IMO, it's a tradeoff between that and our ability to maintain this protocol and be responsible to cover a lot of scenarios.